🔠 Accent: LetterKey +-. hebrew.

ECfaqFck7W opened the pull request:
## Summary
* Split some chars from `VK_COMMA` to shorter `VK_PLUS` & `VK_MINUS`.
	* Adding some relevant chars, like `§` & `|`.
* Adding `GetDefaultLetterKeyHE` for hebrew in `VK_MINUS`

> I hope I opened the request correctly, this is my first time.🙂 I saw that additional points of description were suggested, but I didn`t address them on the assumption that they were not necessary for this simple change.

Thank you!

Issue tracked: https://github.com/microsoft/PowerToys/issues/22711

ECfaqFck7W committed:
🔠 Accent LetterKey add & change

add VK_PLUS & VK_MINUS and change VK_COMMA

{  spaces }

ECfaqFck7W committed:
add GetDefaultLetterKeyHE

VK_MINUS:    { "־", "–" }

modified:   src/modules/poweraccent/PowerAccent.Core/Languages.cs

modified:   src/modules/poweraccent/PowerAccent.Core/Languages.cs

EvyatarEsterson commented:
@microsoft-github-policy-service agree

htcfreek commented:
Do we have an issue that we can link?

damienleroy commented:
Does not work for the french layout keyboards.
Minus/plus are not accessible by this keys. For example minus is on the `VK_6` key.
This is why originally i put them on COMMA key, waiting for a better solution.

It is on your call, but I'm think this tool should not be limited for some layouts.
If it is the case you should precise somewhere it is compatible only for some specific qwerty layouts.

taras-janea commented on src/modules/poweraccent/PowerAccent.Core/Languages.cs:
```diff
@@ -111,7 +113,9 @@ private static string[] GetDefaultLetterKeyALL(LetterKey letter)
                 LetterKey.VK_X => new string[] { "ẋ", "ξ" },
                 LetterKey.VK_Y => new string[] { "ÿ", "ŷ", "ý", "ẏ" },
                 LetterKey.VK_Z => new string[] { "ź", "ž", "ż", "ʒ", "ǯ", "ζ" },
-                LetterKey.VK_COMMA => new string[] { "¿", "¡", "∙", "₋", "⁻", "–", "≤", "≥", "≠", "≈", "≙", "±", "₊", "⁺" },
```
According to @damienleroy, mapping to VK_PLUS, VK_MINUS is not going to work on French keyboards, therefore I'd suggest leaving it mapped to the VK_COMMA symbol as it is.

Evyatar-E replied:
Ok

taras-janea commented on src/modules/poweraccent/PowerAccent.Core/Languages.cs:
```diff
@@ -423,6 +427,16 @@ private static string[] GetDefaultLetterKeyDE(LetterKey letter)
         }
 
+        // Hebrew
+        private static string[] GetDefaultLetterKeyHE(LetterKey letter)
```
Is it a complete set of accents for the Hebrew language?
If not, then please consider either adding these symbols to GetDefaultLetterKey or adding a complete set of Accents for the Hebrew language.

Evyatar-E replied:
I have added some additional less common letters and punctuations, which are unique to Hebrew script. 
Is it worth adding vowels as well?

taras-janea replied:
Yes, if they are accented.

ECfaqFck7W committed:
Revert "🔠 Accent LetterKey add & change"

This reverts commit 83508e80e5543f822a6b94c4d3e930ab6e8ce4cf.

https://github.com/microsoft/PowerToys/pull/22712#discussion_r1048259229

ECfaqFck7W committed:
HE letters and punctuations 2

{ "וֹ", "וּ" }

VK_COMMA

Update Languages.cs
Error SA1028: Code should not contain trailing whitespace

Evyatar-E commented:
@microsoft-github-policy-service agree

jaimecbernardo commented:
Hi @Evyatar-E , is this ready for another review? :)

crutkas commented:
@Evyatar-E ping!  is this ready for another review?

crutkas commented:
@jaimecbernardo can we do the needed changes to help get it across the finish line?

jaimecbernardo committed:
Merge branch 'main' into pr22712

jaimecbernardo commented:
> @jaimecbernardo can we do the needed changes to help get it across the finish line?

Sure, on it.

jaimecbernardo committed:
Add Hebrew to settings

jaimecbernardo commented:
Added the necessary changes to pick the added character set in Settings.

jaimecbernardo reviewed (approved):
LGTM! Thank you for the contribution!
